modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/VisualizationModes.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayLayout from '@clayui/layout';
+import ClayTabs from '@clayui/tabs';
+import React, {ComponentType, useState} from 'react';
+
+import {IFDSViewSectionProps} from '../../FDSView';
+import ListVisualizationMode, {
+	IListVisualizationMode,
+} from './list_visualization_mode/ListVisualizationMode';
+import {
+	ITableVisualizationMode,
+	TableVisualizationMode,
+} from './table_visualization_mode/TableVisualizationMode';
+
+export interface IBaseVisualizationMode<Type extends string> {
+	label: string;
+	type: Type;
+	visualizationModeId: string;
+}
+
+type IVisualizationMode = ITableVisualizationMode | IListVisualizationMode;
```

Comment:
[nit] We commonly name `type`s with `T` prefix, so this would be a `TVisualizationMode`.

Reply:
Sure! I am not used to add prefixes to types :sweat_smile: